uninstall.sh

```diff
@@ -121,9 +121,15 @@ uninstall_from_existing_repositories() {
         return 1
     fi
 
-    find "$START_DIR" -type d -name .git 2>/dev/null | while IFS= read -r EXISTING; do
-        uninstall_hooks_from_repo "$EXISTING"
-    done
+    find "$START_DIR" \( -type d -and -name .git \) -or \
+        \( -type f -and -name HEAD -and -not -path "*/.git/*" \) 2>/dev/null |
+        while IFS= read -r EXISTING; do
+            EXISTING_REPO_ROOT=$(dirname "$EXISTING")
+            if ! is_git_repo "${EXISTING_REPO_ROOT}"; then
```

Comment:
Same as in the install script, this would pass in the directory name, not the path of the directory. /final